### ISSUE TITLE
Updating WCAG H91 question

### ISSUE THREAD

**Issue by fstrr** [w3c/html-aam#452]
Hi

I'm in the process of updating WCAG's [H91 Using HTML form controls and links](https://www.w3.org/WAI/WCAG21/Techniques/html/H91) document to include all the newer input types. I was looking at `input type="color"` which has, using Apple's accessibility inspector, a role of AXColorWell. H91 hasn't listed any AX values, so I'm assuming that it would be better to list IA2_ROLE_COLOR_CHOOSER (but listed as "color chooser"), unless using the Localized Control Type of "color picker" be better.

Any feedback you have would be very useful.

**Comment by JAWS-test** [comment]
The question would be whether we can really recommend input type=color in that case, it's accessibility is rather poor: https://a11ysupport.io/tech/html/input(type-color)_element

**Comment by fstrr** [comment]
Related: I don't see this [color input Safari keyboard bug](https://bugs.webkit.org/show_bug.cgi?id=194756) (that's about to turn 4 years old) listed on that page.

Anyway. Would you say that the table is a recommendation? The preceding sentence is:

> The following table describes how the role, name, value, and state are determined for HTML links and form controls.

My initial question is also relevant to other form elements. For example [input type="date"](https://w3c.github.io/html-aam/#el-input-date) can be a few different things. Which would be best for updating that document?

**Comment by JAWS-test** [comment]
> Would you say that the table is a recommendation?

The table is not a recommendation. But H91 is one. And most of the time it is also right to recommend HTML elements, as they best ensure that the UI elements are correctly perceivable and operable. But unfortunately it is not true for all HTML elements. The question is whether this should be addressed in H91?

**Comment by scottaohara** [comment]
@fstrr, the way that table is described is a bit odd, re: "The following table describes how the role, name, value, and state are determined for HTML links and form controls."

As the role column seems more to indicate how the element MAY be exposed to users - and per your question here, the way these roles are announced may be contingent on the platform or even the browser used when interacting with them.

I'm curious as to how else this table may be getting updated as it seems like there'd be a lot of work that may need to go into it to make it truly accurate to reality, and even if doing so, then it'll likely need to be updated again in the future as things change.  For instance, concerning things like input type=color or file, there are updates that likely need to happen in this spec / with their implementations.   The table from h91 already has inaccuracies / gaps in it.  Maybe a more simplified table that just gets the general concepts right is better than one that goes into each element and has a greater potential of being incorrect as the platform continues to mature?

Note: closing this issue since it's a question, but not an action that needs to be taken with this specification.  please feel free to continue commenting / asking questions though.

**Comment by fstrr** [comment]
A quick note to finish this up: I met with members of AGWG today and we're going to remove the table from that page and link off to external resources instead. Keeping that table up to date would be something of a pyrrhic victory.

Thanks.

**Comment by scottaohara** [comment]
yes.  i absolutely agree that's the better way to go here.
